### PR TITLE
maxima: update to 5.44.0

### PR DESCRIPTION
**Pull request by jbgg**

### CODE REVIEW

File: srcpkgs/maxima/template [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
+# Template file for 'maxima'
 pkgname=maxima
-version=5.43.2
+version=5.44.0
 revision=1
 build_style=gnu-configure
 configure_args="$(vopt_enable clisp) $(vopt_enable sbcl)"
-hostmakedepends="python perl emacs texinfo"
-makedepends="$(vopt_if clisp clisp) $(vopt_if sbcl sbcl)"
+hostmakedepends="python perl emacs texinfo $(vopt_if clisp clisp) $(vopt_if sbcl sbcl)"
```
Comment (Chocimier) [inline srcpkgs/maxima/template:7]: I think it give wrong output from lisp. If you do native build of sbcl, you can likely do native build of maxima too.

**Comment by jbgg** [comment]
When sbcl or clisp is listed on makedepends the build process fails, but instead if sbcl or clisp is listed on hostmakedepends then maxima can be compiled when a cross compilation for musl is done.
I have tested and I can do a native build on musl, but in cross compilation it fails because it is necessary in maxima compilation process.

**Comment by Johnnynator** [comment]
You can add a `closes #19289` to the body of the commit message. So the issue gets closed automatically once the commit is merged.

**Comment by Chocimier** [comment]
Yes, build should fail.

Lisp is called to create architecture-dependent blob. When host list is used, build goes on, but result is incorrect, because it is based on blob for other architecture than target architecture.

There is also no need to do cross build from x86_64 to x86_64-musl, this can be done natively as described in xbps-src readme. What matters is x86_64 to arm.

**Comment by jbgg** [comment]
But, when lisp compiler package is include in makedepends instead of hostmakedepends, why it fails?
Can't be compiled in a musl system (with chroot on masterdir) for cross compiled?

**Comment by Chocimier** [comment]
> But, when lisp compiler package is include in makedepends instead of hostmakedepends, why it fails?

Because makedepends install binaries of target architecture (e.g arm) that are not intended to be run on host architecture (e.g. x86_64).

> Can't be compiled in a musl system (with chroot on masterdir) for cross compiled?


I do not understand your question, can you elaborate?

**Comment by jbgg** [comment]
Sorry, I didn't formulated the question good.

I mean, can xbps-src build a package for x86_64-musl arch on a glibc system host with chroot in masterdir-x86_64-musl?

**Comment by Chocimier** [comment]
Yes it can, documentation on this is updated now.

**Comment by jbgg** [comment]
Thanks!

**Pull request merged**